packages/time-utils/src/index.ts

```
@@ -223,3 +226,49 @@ export function assertDateOnly(date: Date): void {
     `Date is expected to have a no time component (hours/minutes/seconds). Date found: ${date.toISOString()}`
   );
 }
+
+/**
+ * Returns a string with the name of the time units, using the plural form
+ * when necessary.
+ *
+ * @param amount - The number of units.
+ * @param unit - The time unit.
+ */
+export function getTimeUnitLabel(amount: number, unit: TimeUnit): string {
+  switch (unit) {
+    case TimeUnit.MINUTES:
+      return pluralize(amount, "minute", "minutes");
+    case TimeUnit.HOURS:
+      return pluralize(amount, "hour", "hours");
+    case TimeUnit.DAYS:
+      return pluralize(amount, "day", "days");
+    case TimeUnit.WEEKS:
+      return pluralize(amount, "week", "weeks");
+    case TimeUnit.MONTHS:
+      return pluralize(amount, "month", "months");
+    case TimeUnit.YEARS:
+      return pluralize(amount, "year", "years");
+    default:
+      return `${amount} ${unit}`;
```

I think this should just be `return unit;`, right?  Alternatively you could `fail('Unsupported unit: ' + unit);`

---

As an aside, I was tempted to suggest we refactor this to be `getTimePeriodLabel()` and have it return the full "<amount> <unit(s)>" string since it seems like *usually* that will be what you want...  by could imagine maybe needing to style the amount and the units separately or something, so keeping it as `getTimeUnitLabel()` seems reasonable too.  We could also expose both. No strong feelings. 🤷

---

I'm going to leave it for now - I was on the fence too